atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameUnmapperTest.java

```diff
@@ -111,4 +127,82 @@ public void cacheIsActuallyUsed() throws TableMappingNotFoundException {
         verify(row, times(1)).getString("short_table_name");
     }
+
+    @Test
+    public void getLongTableNamesFromMappingTableReturnsLongNames() throws TableMappingNotFoundException {
+        ArgumentCaptor<Object> tableNameCaptor = ArgumentCaptor.forClass(Object.class);
+        Map<String, String> shortNamesToLongNames = ImmutableMap.<String, String>builder()
+                .put("shortNameOne", "superLongNameOne")
+                .put("shortNameTwo", "superLongNameTwo")
+                .put("shortNameThree", "superLongNameThree")
+                .build();
+        mockShortNamesToLongNamesQuery(shortNamesToLongNames.values());
+
+        Set<String> longNames = oracleTableNameUnmapper.getLongTableNamesFromMappingTable(
+                connectionSupplier, shortNamesToLongNames.keySet());
+
+        assertThat(longNames).containsExactlyInAnyOrderElementsOf(shortNamesToLongNames.values());
+        verify(sqlConnection)
+                .selectResultSetUnregisteredQuery(
+                        eq("SELECT table_name FROM atlasdb_table_names WHERE LOWER(short_table_name) IN ("
+                                + generatePlaceholders(3) + ")"),
+                        tableNameCaptor.capture());
+
+        assertThat(tableNameCaptor.getAllValues()).containsExactlyInAnyOrderElementsOf(shortNamesToLongNames.keySet());
+        verifyNoMoreInteractions(sqlConnection);
+    }
+
+    @Test
+    public void getLongTableNamesFromMappingTableDoesMultipleQueriesIfMoreTablesThanThreshold()
+            throws TableMappingNotFoundException {
+        int numberOfEntries = AtlasDbConstants.IN_CLAUSE_EXPRESSION_LIMIT + 100;
+        Map<String, String> shortNamesToLongNames = IntStream.range(0, numberOfEntries)
+                .boxed()
+                .collect(Collectors.toMap(i -> "shortName" + i, i -> "longName" + i));
+        mockShortNamesToLongNamesQuery(shortNamesToLongNames.values());
+
+        Set<String> longNames = oracleTableNameUnmapper.getLongTableNamesFromMappingTable(
+                connectionSupplier, shortNamesToLongNames.keySet());
+
+        assertThat(longNames).containsExactlyInAnyOrderElementsOf(shortNamesToLongNames.values());
+        verify(sqlConnection)
+                .selectResultSetUnregisteredQuery(
+                        eq("SELECT table_name FROM atlasdb_table_names WHERE LOWER(short_table_name) IN ("
+                                + generatePlaceholders(AtlasDbConstants.IN_CLAUSE_EXPRESSION_LIMIT) + ")"),
+                        any());
+        verify(sqlConnection)
+                .selectResultSetUnregisteredQuery(
+                        eq("SELECT table_name FROM atlasdb_table_names WHERE LOWER(short_table_name) IN ("
+                                + generatePlaceholders(numberOfEntries - AtlasDbConstants.IN_CLAUSE_EXPRESSION_LIMIT)
+                                + ")"),
+                        any());
+        verifyNoMoreInteractions(sqlConnection);
+    }
```

Comment:
Yeah, I added it since I wanted to convince myself that we weren't just doing even more requests. While changing the batching is not incorrect, it'll fail this test if there are more requests (well, actually it'll fail if the number of connections doesn't change but the spread of placeholders changes) which I guess makes you think if you really want N many more requests